### ISSUE TITLE
bogus SC2086 (no support for bats run or shell exec?)

### ISSUE THREAD

**Issue by kolyshkin** [koalaman/shellcheck#2068]
#### For bugs
- Rule Id (if any, e.g. SC1000): `SC2086`
- My shellcheck version (`shellcheck --version` or "online"): `0.7.1`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/env bats

@test xexe {
	ECHO_BIN=/bin/echo
	ECHO_FLAGS=-e
	ECHO="$ECHO_BIN $ECHO_FLAGS"

	run $ECHO '\t\twow\t\towo'
	false
}
```

#### Here's what shellcheck currently says:

```
In a.bats line 8:
	run $ECHO '\t\twow\t\towo'
            ^---^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
	run "$ECHO" '\t\twow\t\towo'

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

#### Here's what I wanted or expected to see:

No warning.

If I run the same code in pure bash:

```sh
#!/bin/env bash

ECHO_BIN=/bin/echo
ECHO_FLAGS=-e
ECHO="$ECHO_BIN $ECHO_FLAGS"

$ECHO '\t\twow\t\towo'
```

I have no warnings. Although if I add `exec` it appears:

```
In ./a.bash line 7:
exec $ECHO '\t\twow\t\towo'
     ^---^ SC2086: Double quote to prevent globbing and word splitting.
```

It looks like `run` (which is a built-in bats feature to execute a command, placing its stdout+stderr to `$output` and its exit status to `$status`) is not recognized.

Maybe I'm wrong here but it seems beneficial to support some reserved words (`exec` for bash and bats, and `run` for bats) to decrease the number of bogus warnings.

**Comment by koalaman** [comment]
The check for whether the unquoted token was used as a command name now correctly recognizes `exec` & friends, and these now include `run`. Thanks!
